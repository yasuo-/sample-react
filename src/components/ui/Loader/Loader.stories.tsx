import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Loader } from './Loader';

export default {
  title: 'Components/UI/Loader',
  component: Loader,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Loader>;

const Template: ComponentStory<typeof Loader> = (args) => (
  <div style={{ width: '100px', height: '100px' }}>
    <Loader {...args} />
  </div>
);

export const DefaultSize = Template.bind({});
export const SmallSize = Template.bind({});
export const LargeSize = Template.bind({});

DefaultSize.args = {
  size: 'large',
};

SmallSize.args = {
  size: 'small',
};

LargeSize.args = {
  size: 'small',
};
