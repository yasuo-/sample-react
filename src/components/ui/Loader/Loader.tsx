import React from 'react';

import { Loader as AwsLoader, type LoaderProps as AwsLoaderProps } from '@aws-amplify/ui-react';

type LoaderProps = AwsLoaderProps;

export const Loader: React.FC<LoaderProps> = ({ size = 'large', isDeterminate, percentage }) => (
  <AwsLoader size={size} percentage={percentage} isDeterminate={isDeterminate} />
);

Loader.displayName = 'Loader';
