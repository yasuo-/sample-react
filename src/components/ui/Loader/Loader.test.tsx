import React from 'react';

import { render } from '@testing-library/react';

import { Loader } from './Loader';

describe('Loader Ui Components', () => {
  it('displayName is Loader', () => {
    expect(Loader.displayName).toBe('Loader');
  });

  it('should snapshot', () => {
    const { asFragment } = render(
      <div style={{ width: '100px', height: '100px' }}>
        <Loader />
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have role img(svg)', () => {
    const rendered = render(<Loader size={'large'} />);
    expect(rendered.findByRole('img')).toBeDefined();
  });
});
