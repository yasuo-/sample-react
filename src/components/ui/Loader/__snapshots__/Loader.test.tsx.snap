// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Ui Components should snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100px; height: 100px;"
  >
    <svg
      class="amplify-loader amplify-loader--large"
      data-size="large"
      role="img"
    >
      <g>
        <circle
          cx="50%"
          cy="50%"
          data-testid="circular-empty"
          r="42%"
          stroke-width="8%"
          style="stroke: undefined;"
        />
        <circle
          cx="50%"
          cy="50%"
          data-testid="circular-filled"
          r="42%"
          stroke-width="8%"
          style="stroke: undefined;"
        />
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
